File: BackEnd/src/domain/interfaces/usecases/user/IUpdateUserProfileUseCase.ts
/**
 * @file IUpdateUserProfileUseCase.ts
 * @description Interfaz del caso de uso para actualizar el perfil del usuario.
 * @module User
 * @layer Domain
 */

import type { UpdateProfileRequestDto } from '../../../dtos/user/update-profile-request.dto.js';
import type { UserResponseDTO } from '../../../dtos/user/user-response.dto.js';

export interface IUpdateUserProfileUseCase {
  /**
   * @param userId — ID del usuario autenticado.
   * @param dto — Campos a actualizar (nombre, bio, ubicación, foto).
   * @returns Datos públicos del usuario actualizado.
   * @throws {NotFoundException} Si el usuario no existe.
   */
  execute(userId: string, dto: UpdateProfileRequestDto): Promise<UserResponseDTO>;
}

File: FrontEnd/lib/domain/interfaces/usecases/user/i_update_user_profile_use_case.dart
/// @file i_update_user_profile_use_case.dart
/// @description Interfaz: Actualiza los datos de perfil del usuario.
/// @module User
/// @layer Domain
library;

import '../../../../domain/dtos/user/update_profile_request_dto.dart';
import '../../../../domain/entities/user.dart';

abstract interface class IUpdateUserProfileUseCase {
  /// Actualiza los datos de perfil del usuario autenticado.
  ///
  /// [param] dto — campos a actualizar (todos opcionales).
  /// [returns] [User] con los datos actualizados.
  Future<User> execute(UpdateProfileRequestDto dto);
}
